modules/pel/peltool/peltool.py

```diff
@@ -267,7 +267,84 @@ def parseAndWriteOutput(file: str, output_dir: str, config: Config,
 
 
+def parsePELSummary(stream: DataStream, config: Config, critSysTerm:bool = False):
+    """
+    Parses and summarizes data from a PEL stream.
+    Returns: Event ID (eid) and summary of PEL data.
+            If no PEL is parsed or an error occurs, empty strings are returned.
+    """
+    out = OrderedDict()
+    ret, ph = generatePH(stream, out)
+    if ret is False:
+        return "", ""
+    eid = ph.pLID
+    ret, uh = generateUH(stream, ph.creatorID, out)
+    if ret is False:
+        return "", ""
+    if not uh.isServiceable():
+        return "", ""
+    
+    summary = OrderedDict()
+    for _ in range(2, ph.sectionCount):
+        sectionID, sectionLen, versionID, subType, componentID = parseHeader(stream)
+        section_json = OrderedDict()
+        sectionFun(stream, section_json, sectionID, sectionLen,
+                   versionID, subType, componentID, ph.creatorID, config)
+        if sectionID == SectionID.primarySRC.value:
+            summary["SRC"] = section_json["Primary SRC"]["Reference Code"]
+            if "Error Details" in section_json["Primary SRC"] :
+                summary["Message"] = section_json["Primary SRC"]["Error Details"]["Message"]
+            break
+    summary["PLID"] = ph.pLID
+    summary["CreatorID"] = out["Private Header"]["Creator Subsystem"]
+    summary["Subsystem"] = out["User Header"]["Subsystem"]
+    summary["Commit Time"] = ph.commitTime
+    summary["Sev"] = out["User Header"]["Event Severity"]
+    summary["CompID"] = out["Private Header"]["Created by"]
+    return eid, summary
+
+def extractAndSummarizePEL(file: str, config: Config,
+                        delete_after_parsing: bool):
+    """
+    Extracts and summarizes data from a PEL file.
+    Returns: Event ID (eid) and summary extracted from the PEL.
+            If no PEL is parsed, empty strings are returned.
+    """
+    with open(file, 'rb') as fd:
+        data = fd.read()
+        stream = DataStream(data, byte_order='big', is_signed=False)
+        try:
+            eid, summary = parsePELSummary(stream, config)
+            if eid :
+                return eid, summary
+        except Exception as e:
+            print(f"Exception: No PEL parsed for {file}: {e}")
+    return "", ""
+
+def listOption(path: str, config: Config, extension: str, delete: bool, rev: bool =False):
+    file_list = []
+    root = ""
+    for root, _, files in os.walk(path):
+        for file in files:
+            if extension and extension != os.path.splitext(file)[1]:
+                continue
+            file_list.append(file)  ## create list of file names
+        # Only process top level directory
+        break
+    file_list.sort(reverse=rev) 
+    final_summary = {}
+    for file in file_list:
+        eid, summary = extractAndSummarizePEL(os.path.join(root, file), config, delete)
+        if eid :
+            final_summary[eid] = summary
+    print(prettyPrint(json.dumps(final_summary, indent=4)))
```

Comment:
there's a bit more of an indent with this than with the original peltool:

original:
```
"SRC":                  "110015F0",
"Message":              "A power suppl
```
this one:
```
"SRC":                       "BD503001",
"Message":                   "Failure occurred during boot process",
```

not a huge deal but is there a way to remove some spaces (looks like its 5) from this output?

Reply:
looks good!